gap/Utils.gi

```diff
@@ -354,3 +354,16 @@ function(epsilon, n, q)
     
     return rec(generatorsOfSO := generatorsOfSO, D := D, E := E);
 end);
+
+InstallGlobalFunction("MatrixGroup",
+function(gens, F)
+    return Group(List(gens, g -> ImmutableMatrix(F, g));
```

Comment:
I don't think this has any effect *right now* compared to just `Group(gens)` which already copies all generators into immutable ones and compresses them if possible. Now in the future this may change but only if using the new MatrixObj API, which you largely don't.